environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceEncryptionInitializationHandler.java

```diff
@@ -58,10 +88,18 @@ public void accept(Event<EnvironmentDto> environmentDtoEvent) {
                             .map(azureParamsDto -> azureParamsDto.getAzureResourceEncryptionParametersDto())
                             .map(azureREParamsDto -> azureREParamsDto.getEncryptionKeyUrl()).orElse(null);
                     if (StringUtils.isNotEmpty(encryptionKeyUrl)) {
-                        LOGGER.debug("Creating DiskEncryptionSet for environment.");
-                            /*
-                            TO DO - create the Disk Encryption Set and save environment.
-                            */
+                        LOGGER.debug("Validating if encryptionKeyUrl exists in Azure cloud.");
+                        boolean keyExistsInCloud = environmentEncryptionService.validateEncryptionParameters(encryptionKeyUrl);
+                        if (keyExistsInCloud) {
+                            LOGGER.debug("{} exists in Aure cloud, Creating DiskEncryptionSet for environment.", encryptionKeyUrl);
+
+                            String resourceGroupName = (String) getOrCreateResourceGroup(environmentDto, environment);
+                            String diskEncryptionSetUrl = environmentEncryptionService.createEncryptionSet(encryptionKeyUrl, resourceGroupName);
+                            environment.setEnvironmentEncryptionParameters(diskEncryptionSetUrl);
```

Comment:
The placement and name of this property is misleading and unclear as mentioned above.